pkg/gitrepo/repo.go

```diff
@@ -36,6 +38,11 @@ type Repo struct {
 	worktree billy.Filesystem
 }
 
+type CheckoutOptions struct {
+	Branch string
+	Tag    string
```

Comment:
I also added `Tag` because this make testing easier, since tag should not change as opposed to branches which change oftenly.